src/Castle.Windsor.Extensions.DependencyInjection/WindsorScopedServiceProvider.cs

```diff
@@ -27,48 +27,40 @@ namespace Castle.Windsor.Extensions.DependencyInjection
 	internal class WindsorScopedServiceProvider : IServiceProvider, ISupportRequiredService, IDisposable
 	{
 		private readonly ExtensionContainerScope scope;
-		private bool disposing = false;
+		private bool disposing;
 
 		private readonly IWindsorContainer container;
 		
 		public WindsorScopedServiceProvider(IWindsorContainer container)
 		{
 			this.container = container;
-			this.scope = ExtensionContainerScope.Current;
+			scope = ExtensionContainerScope.Current;
 		}
 
 		public object GetService(Type serviceType)
 		{
-			using(var fs = new ExtensionContainerScope.ForcedScope(scope))
+			using(var _ = new ForcedScope(scope))
 			{
 				return ResolveInstanceOrNull(serviceType, true);	
 			}
 		}
 
 		public object GetRequiredService(Type serviceType)
 		{
-			using(var fs = new ExtensionContainerScope.ForcedScope(scope))
+			using(var _ = new ForcedScope(scope))
 			{
 				return ResolveInstanceOrNull(serviceType, false);	
 			}
 		}
 
 		public void Dispose()
 		{
-			if(scope is ExtensionContainerRootScope)
-			{
-				if(!disposing)
-				{
-					disposing = true;
-					var disposableScope = scope as IDisposable;
-					if(disposableScope != null)
-					{
-						disposableScope.Dispose();
-					}
-					container.Dispose();
-				}
-				
-			}
+			if (!(scope is ExtensionContainerRootScope)) return;
+			if (disposing) return;
+			disposing = true;
+			var disposableScope = scope as IDisposable;
+			disposableScope?.Dispose();
+			container.Dispose();
```

Comment:
I know this was there already before your pull request but it surprised me. Shouldn't it be the job of the person who created the container to dispose it, what if they wanted to continue using it. If there is a good reason for it having to work like this could we add a comment explaining that here or log a task to investigate.

Reply:
Yes i though the same thing when i refactored.
**I think** the reason is that the Provider is disposed with the host, and this is the reason for the container being disposed here.

Reply:
I.e. this is needed for the container to be dispoed when the Hosts stops. However i can do two thinks (maybe a seperate Issue):
- Add an option to the Provider Factory that disableds the automatic dispose of the container with the provider
- And, Or, make the disposable virtual to allow an override for the disposiing.
I will add an issue. Should not block this PR.

Reply:
Yer, just log a new issue to discuss it there to work out if it is a code change, code comments change or docs change since the docs don't mention you give ownership of the container over when you pass it.

Reply:
Please see change